src/libsync/discovery.cpp

```diff
@@ -0,0 +1,1218 @@
+                    processRename(path);
+                } else if (isMove) {
+                    // We must query the server to know if the etag has not changed
+                    _pendingAsyncJobs++;
+                    QString serverOriginalPath = originalPath;
+                    if (localEntry.isVirtualFile)
+                        serverOriginalPath.chop(_discoveryData->_syncOptions._virtualFileSuffix.size());
+                    auto job = new RequestEtagJob(_discoveryData->_account, serverOriginalPath, this);
+                    connect(job, &RequestEtagJob::finishedWithResult, this, [=](const Result<QString> &etag) mutable {
+                        if (!etag || (*etag != base._etag && !item->isDirectory()) || _discoveryData->_renamedItems.contains(originalPath)) {
+                            qCInfo(lcDisco) << "Can't rename because the etag has changed or the directory is gone" << originalPath;
+                            // Can't be a rename, leave it as a new.
+                            postProcessLocalNew();
+                        } else {
+                            // In case the deleted item was discovered in parallel
+                            auto it = _discoveryData->_deletedItem.find(originalPath);
+                            if (it != _discoveryData->_deletedItem.end()) {
+                                ASSERT((*it)->_instruction == CSYNC_INSTRUCTION_REMOVE);
+                                (*it)->_instruction = CSYNC_INSTRUCTION_NONE;
+                            }
+                            delete _discoveryData->_queuedDeletedDirectories.take(originalPath);
+
+                            processRename(path);
+                            recurseQueryServer = *etag == base._etag ? ParentNotChanged : NormalQuery;
+                        }
+                        processFileFinalize(item, path, item->isDirectory(), NormalQuery, recurseQueryServer);
+                        _pendingAsyncJobs--;
+                        QTimer::singleShot(0, _discoveryData, &DiscoveryPhase::scheduleMoreJobs);
+                    });
+                    job->start();
+                    return;
+                } else {
+                    postProcessLocalNew();
+                }
+            } else {
+                postProcessLocalNew();
+            }
+        } else {
+            item->_instruction = CSYNC_INSTRUCTION_SYNC;
+            if (noServerEntry) {
+                // Special case! deleted on server, modified on client, the instruction is then NEW
+                item->_instruction = CSYNC_INSTRUCTION_NEW;
+            }
+            item->_direction = SyncFileItem::Up;
+            item->_checksumHeader.clear();
+            item->_size = localEntry.size;
+            item->_modtime = localEntry.modtime;
+            item->_previousSize = dbEntry._fileSize;
+            item->_previousModtime = dbEntry._modtime;
+            if (!_dirItem || _dirItem->_direction == SyncFileItem::Down) {
```

Comment:
There have been more of these along the way, but I still don't understand

Reply:
this condition no longer exists